src/test.py

```diff
@@ -16,66 +16,114 @@
 # Assume 4 players and initialized weights for all of them regardless of whether they are AI or human
 # All are initialized with a placeholder dict that should be overwritten
 for i in range(4):
-    weightLogs[i] = Log('../logs/WeightLog_%s.txt' % i)
+    weightLogs[i] = Log('../logs/qweights_new_%s.txt' % i)
     # weightLogs[i].log_dict({'DELETE ME': -1})
 
 winners = {}  # Map from the player turn index to the number of wins they have had
-runs = 1
+runs = 2000
```

Comment:
Is it an idea to write your own test.py and call it QLearning test.py, that way when you want to train this you can call your own, and we can still train the other one and see how they perform against each other? Just a thought, if it is too much work then don't bother.

Reply:
+1. I think as we're doing these learning updates, we should keep them in separate files so that we can refer back to them and leave more room for experimentation.

Good work!